### PR TITLE
Add common spec to dependencies

### LINKED ISSUE #635: Spark-dependencies how to specify the nodeSelector option
Spark-dependencies requires a lot of memory, I want to run on the specified node.
```shell
Exception in thread "main" org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1.0 (TID 5, localhost, executor driver): java.lang.OutOfMemoryError: GC overhead limit exceeded
	at java.lang.AbstractStringBuilder.<init>(AbstractStringBuilder.java:68)
	at java.lang.StringBuilder.<init>(StringBuilder.java:89)
	at java.io.ObjectInputStream$BlockDataInputStream.readUTFBody(ObjectInputStream.java:3404)
	at java.io.ObjectInputStream$BlockDataInputStream.readUTF(ObjectInputStream.java:3224)
	at java.io.ObjectInputStream.readString(ObjectInputStream.java:1903)
	at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1562)
	at java.io.ObjectInputStream.readObject(ObjectInputStream.java:431)
	at org.apache.spark.serializer.JavaDeserializationStream.readObject(JavaSerializer.scala:75)
	at org.apache.spark.serializer.DeserializationStream.readKey(Serializer.scala:156)
	at org.apache.spark.serializer.DeserializationStream$$anon$2.getNext(Serializer.scala:188)
	at org.apache.spark.serializer.DeserializationStream$$anon$2.getNext(Serializer.scala:185)
	at org.apache.spark.util.NextIterator.hasNext(NextIterator.scala:73)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:438)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:153)
	at org.apache.spark.Aggregator.combineValuesByKey(Aggregator.scala:41)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:90)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
```

### PR DESCRIPTION
**Pull request by objectiser**
Signed-off-by: Gary Brown <gary@brownuk.com>

Fixes #635 

Relates to #414

### CODE REVIEW

File: pkg/apis/jaegertracing/v1/jaeger_types.go [hunk 1]
```diff
@@ -378,7 +378,7 @@ type JaegerDependenciesSpec struct {
 	TTLSecondsAfterFinished *int32 `json:"ttlSecondsAfterFinished,omitempty"`
 
 	// +optional
-	Resources v1.ResourceRequirements `json:"resources,omitempty"`
```
Comment (jpkrohling) [inline pkg/apis/jaegertracing/v1/jaeger_types.go]: This breaks compatibility with CRs specifying that node, doesn't it? Would it be possible to deprecate first, and use the value from the CR to populate the new property?

File: pkg/cronjob/spark_dependencies_test.go [hunk 2]
```diff
@@ -131,7 +170,9 @@ func TestSparkDependenciesResources(t *testing.T) {
 				Storage: v1.JaegerStorageSpec{
 					Type: "elasticsearch",
 					Dependencies: v1.JaegerDependenciesSpec{
-						Resources: dependencyResources,
```
Comment (jpkrohling) [inline pkg/cronjob/spark_dependencies_test.go]: If we go for a backwards-compatible change, can you keep this test, adding a new case for the new property?

File: pkg/apis/jaegertracing/v1/jaeger_types.go (hunk 1 above)
Comment (objectiser) [reply pkg/apis/jaegertracing/v1/jaeger_types.go]: The `commonSpec.Resources` is a direct replacement as the `CommonSpec` is inline. So from the CR's perspective there is no change.

File: pkg/cronjob/spark_dependencies_test.go (hunk 2 above)
Comment (objectiser) [reply pkg/cronjob/spark_dependencies_test.go]: It would only not be backward compatible for other Go apps (I guess operators) reusing the jaeger-types?

**Pull request merged**